src/js/game/components/processing_requirement.js

```diff
@@ -0,0 +1,90 @@
+import { Component } from "../component";
+import { enumItemProcessorTypes } from "./item_processor";
+import { Entity } from "../entity";
+import { BOOL_TRUE_SINGLETON } from "../items/boolean_item";
+import { BaseItem } from "../base_item";
+import { ShapeItem } from "../items/shape_item";
+
+export class ProcessingRequirementComponent extends Component {
+    static getId() {
+        return "ProcessingRequirement";
+    }
+
+    duplicateWithoutContents() {
+        return new ProcessingRequirementComponent({
+            processorType: this.type
+        });
+    }
+
+    /**
+     *
+     * @param {object} param0
+     * @param {enumItemProcessorTypes=} param0.processorType Which type of processor this is
+     *
+     */
+    constructor({ processorType = enumItemProcessorTypes.painterQuad }) {
+        super();
+
+        // Type of the processor
+        this.type = processorType;
+    }
+
+    /**
+     * Checks whether it's possible to process something
+     * @param {Entity} entity
+     */
+    canProcess(entity) {
```

Comment:
Hm, I don't like that there is so much logic in this method. Could it be moved into a game system?

Reply:
moved to itemprocessor system as separate method